### ISSUE TITLE
gem "bundler" cannot be uninstalled because it is a default gem

### ISSUE THREAD

**Issue by willmanio** [FutureWorkshops/bitrise-step-install-bundler#3]
Running this step on the _Xcode 10.0.x on macOS 10.13 (High Sierra)_ stack, with no other modifications to the set of installed software or workspace.

`gem --version`: `2.7.7`
`ruby --version`: `ruby 2.5.1p57 (2018-03-29 revision 63029) [x86_64-darwin17]`
`bundler --version`: `Bundler version 1.16.2`

The step is unable to uninstall the existing version of Bundler. I'm not sure this is actually necessary. There are two use cases:

- Installing Bundler 2.0 on a Bitrise stack that only includes 1.0
- Installing Bundler 1.0 on a Bitrise stack that only includes 2.0

Both versions of Bundler can coincide peacefully. In fact 2.0 contains a [Version autoswitch](https://bundler.io/guides/bundler_2_upgrade.html#version-autoswitch) functionality for the second use case. I think it would be sufficient to parse `Gemfile.lock` to find the appropriate Bundler version and then `gem install` that specific version.

I'm able to work around this issue with a Script step which invokes `gem install bundler -v "2.0.1" --force` but I'm sure there's something more elegant than the `--force` flag.

See the full log output.
```
+------------------------------------------------------------------------------+

| (3) install-bundler@1.0.0                                                    |
+------------------------------------------------------------------------------+
| id: install-bundler                                                          |
| version: 1.0.0                                                               |
| collection: https://github.com/bitrise-io/bitrise-steplib.git                |
| toolkit: bash                                                                |
| time: 2019-02-25T12:21:27-08:00                                              |
+------------------------------------------------------------------------------+
|                                                                              |
Configs:
  * gemfile_path: /Users/vagrant/git/ios/Gemfile.lock
Could not locate Gemfile
Current bundler can't be used to run this project. Updating...
Gemfile expected version: 2.0.1
Current reported version: 1.16.2
Uninstalling current bundler
ERROR:  While executing gem ... (Gem::InstallError)
    gem "bundler" cannot be uninstalled because it is a default gem
|                                                                              |
+---+---------------------------------------------------------------+----------+
| x | install-bundler@1.0.0 (exit code: 1)                          | 3.91 sec |
+---+---------------------------------------------------------------+----------+
| Issue tracker: ...ub.com/FutureWorkshops/bitrise-step-install-bundler/issues |
| Source: https://github.com/FutureWorkshops/bitrise-step-install-bundler.git  |
+---+---------------------------------------------------------------+----------+
```

**Comment by kesavaj** [comment]
Did you get to solve this ?

**Comment by willmanio** [comment]
Nope, I ended up adding a script step with the following:

```
#!/usr/bin/env bash -ex

# This should be replaced with the "Install specific bundler version" step once some outstanding bugs are fixed
# See https://github.com/FutureWorkshops/bitrise-step-install-bundler

gem install bundler -v "2.0.1" --force
```

Here's the relevant section from our bitrise.yml:


```
    - script@1.1.5:
        inputs:
        - content: |-
            #!/usr/bin/env bash -ex

            # This should be replaced with the "Install specific bundler version" step once some outstanding bugs are fixed
            # See https://github.com/FutureWorkshops/bitrise-step-install-bundler

            gem install bundler -v "2.0.1" --force
        title: Install Bundler 2.0.1
```

**Comment by willmanio** [comment]
PS, if FutureWorkshops does not fix these outstanding issues, then I plan on forking this plugin when I have some free time.

**Comment by v-andr** [comment]
script worked as a charm for me as well, thanks @willmanio 
the downside is the need to update it every time bundler version changes, but it shouldn't be that often

**Comment by igorcferreira** [comment]
@willmanio Sorry for the long time to respond for your issue. I'll be observing the Future Workshops' build steps more carefully now. I have a question: Does the code merged on PR #4 solves the need for this extra step?
